swarm/src/handler.rs

```diff
@@ -125,25 +124,23 @@ pub trait ConnectionHandler: Send + 'static {
 
     /// Returns until when the connection should be kept alive.
```

Comment:
I think we need to completely overhaul this together with https://github.com/libp2p/rust-libp2p/pull/4595. What do you think of merging this PR into https://github.com/libp2p/rust-libp2p/pull/4595? It is so closely related, I think it makes sense to just merge it as a single-patch.

Reply:
Sure, what do your think we also change `KeepAlive` to boolean in #4595

Reply:
> Sure, what do your think we also change `KeepAlive` to boolean in #4595

I think it makes sense. With only two variants, we don't really gain anything from having another type and I cannot see how somebody could misinterpret what `true` and `false` mean in this case.